### PR TITLE
feat: add mobile hamburger menu

### PR DESCRIPTION
**Pull request by BruceWangyq**
- add main nav and mobile nav components 
- add header responsive
- add types/index.ts

### CODE REVIEW

File: hooks/use-lock-body.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// @see https://usehooks.com/useLockBodyScroll.
+export function useLockBody() {
+  React.useLayoutEffect((): (() => void) => {
+    const originalStyle: string = window.getComputedStyle(
+      document.body
+    ).overflow
+    document.body.style.overflow = "hidden"
+    return () => (document.body.style.overflow = originalStyle)
+  }, [])
+}
```
Comment (PeerRich) [inline hooks/use-lock-body.ts:12]: TIL, nice hook!

File: src/components/mobile-nav.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          type="text"
+          placeholder="Search or jump to…"
+        />
+        <nav className="grid grid-flow-row auto-rows-max text-sm">
+          {items.map((item, index) => (
+            <Link
+              key={index}
+              href={item.disabled ? "#" : item.href}
+              className={cn(
+                "flex w-full items-center border-b border-b-slate-200 p-3 text-sm font-medium hover:text-gray-400 ",
```
Comment (PeerRich) [inline src/components/mobile-nav.tsx:36]:
```suggestion
                "flex w-full items-center border-b border-b-border-[#383B42] p-3 text-sm font-medium hover:text-gray-400 ",
```

Comment (PeerRich) [reply src/components/mobile-nav.tsx:36]: we should add 383B42 into a tailwind theme color at some point

Comment (BruceWangyq) [reply src/components/mobile-nav.tsx:36]: Will change that!

Comment (BruceWangyq) [reply src/components/mobile-nav.tsx]: > we should add 383B42 into a tailwind theme color at some point
like this?

module.exports = {
  theme: {
    colors: {
      'border': '#383B42',
    },
  },
}

**Comment by BruceWangyq** [comment]
Should be good now, added customized color to the tailwind config.

File: hooks/use-lock-body.ts (hunk 1 above)
Comment (BruceWangyq) [reply hooks/use-lock-body.ts]: Learned this from shadcn

File: .gitignore [hunk 3]
```diff
@@ -40,3 +40,6 @@ yarn-error.log*
 
 # typescript
 *.tsbuildinfo
+
+#ide 
+.vscode/*
```
Comment (pierresisson) [inline .gitignore:45]: prettier

File: next.config.mjs [hunk 4]
```diff
@@ -16,9 +19,9 @@ const config = {
...
    * @see https://github.com/vercel/next.js/issues/41980
    */
-  i18n: {
-    locales: ["en"],
-    defaultLocale: "en",
-  },
+  // i18n: {
+  //   locales: ["en"],
+  //   defaultLocale: "en",
+  // },
```
Comment (pierresisson) [inline next.config.mjs:25]: remove comment because we do not need them atm ?

**Review by pierresisson — APPROVED** [review]
LGTM nice PR

**Comment by cypherhoodlum** [comment]
Is this ready to be merged as soon as @BruceWangyq resolves the conflicts in src/components/ui/header.tsx? Good job btw! :rocket:

File: next.config.mjs (hunk 4 above)
Comment (BruceWangyq) [reply next.config.mjs]: Done!

**Comment by PeerRich** [comment]
just noticed: the border colors look odd

**Comment by BruceWangyq** [comment]
> just noticed: the border colors look odd



> Yeah, I will fix that when I solve the merge conflict

**Review by PeerRich — APPROVED** [review]
looks great!

**Pull request merged**